src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -264,26 +312,43 @@ bool mongoConnect(const char*         host,
     std::string versionString = std::string(result.getStringField("version"));
     if (!versionParse(versionString, mongoVersionMayor, mongoVersionMinor, extra))
     {
-        mongoSemGive(__FUNCTION__, "wrong mongo version format");
         LM_E(("Database Startup Error (invalid version format: %s)", versionString.c_str()));
-        return false;
+        return NULL;
     }
     LM_T(LmtMongo, ("mongo version server: %s (mayor: %d, minor: %d, extra: %s)", versionString.c_str(), mongoVersionMayor, mongoVersionMinor, extra.c_str()));
 
-    mongoSemGive(__FUNCTION__, "connecting to mongo");
-    return true;
+    return connection;
 }
 
+
+
 /* ****************************************************************************
 *
 * mongoConnect -
 *
-* Version of the functions that doesn't uses authentication parameters
+* Version of the function that doesn't use authentication parameters.
+* FIXME: no longer used?
 */
-bool mongoConnect(const char* host) {
+bool mongoConnect(const char* host)
+{
+  return true;
+}
+
 
-    return mongoConnect(host, "", "", "", "", false, 0);
+#ifdef UNIT_TEST
```

Comment:
Maybe this block should under unit test code? Or it would be hard doing in that way?